### DISCUSSION TITLE
Can I adjust the number of replicas between blue/green deployments?

### DISCUSSION THREAD

**Discussion by hoiheart** [argoproj/argo-rollouts#1267] [Q&A] [answered]
Hello

Can I adjust the number of replicas between blue/green deployments?

If the replica is set to 5, it increases to 10 between blue/green deployments.

This can lead to resource problems in situations where nodes are limited.

For example, is there a way to adjust a blue/green deployment between deployments to active 3, preview 2 and increase it back to 5 after deployment is complete?

I need your help.

Thanks.

**Comment by nextrevision** [comment, marked as answer]
There's the ability to set the preview replica count with `spec.strategy.blueGreen.previewReplicaCount`. Could you set in your manifest the following:

```
...
spec:
  replicas: 3
  strategy:
    blueGreen:
      previewReplicaCount: 2
...
```

Then when the promotion is complete, run `kubectl scale --replicas=5 rollout/myrollout`? Optionally, I suppose you can adjust the scale (using the same command) down to 3 before ever applying the manifest update.
